airbyte-cdk/python/airbyte_cdk/sources/streams/http/requests_native_auth/abstract_oauth.py

```diff
@@ -29,10 +29,9 @@ def get_auth_header(self) -> Mapping[str, Any]:
     def get_access_token(self) -> str:
         """Returns the access token"""
         if self.token_has_expired():
-            t0 = pendulum.now()
```

Comment:
one slight drawback to moving the `now()` generation into the authenticator set token expiry methods is that we do have a very small corner case where we may attempt a request with a token that has expired.

When we were using `pendulum.now()` before refreshing the access token, the corner case meant we might think a token is already expired and refresh it earlier than necessary. 

However, now with the reverse of calling `pendulum.now()` after already refreshing the token, then at the very end of expiry, we may attempt a request where the token has already expired.

I'd argue that refreshing the token a little more often than necessary is better than attempting requests with an already expired token so just to be on the safe side, I think we should continue to get the current time before calling `self.refresh_access_token()` and pass it into `self.set_token_expiry_date()` Its a little verbose to pass both params in, but we're already changing the interface regardless.